lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1418 @@
+        .then(() => driver.enterRemoteCommandMode())
+        .then(() => driver.togglePassthroughMode(BCNLDriver.PASSTHROUGH_MODE.ENABLE))
+        .then(() => driver.openConnection())
+        .then(() => driver.readPumpInfo())
+        .then(() => driver.getLinkKey())
+        .then(() => driver.negotiateRadioChannel())
+        .then(() => driver.toggleHighSpeedMode(HighSpeedModeCommand.HIGH_SPEED_MODE.ENABLE))
+        .then(() => driver.getPumpTime())
+        .then((pumpTime) => {
+          settings.pumpTime = sundial.applyTimezone(pumpTime.time, cfg.timezone);
+        })
+        .then(() => driver.getPumpStatus())
+        .then((status) => {
+          settings.activeBasalPattern =
+            MM600SeriesDriver.PATTERN_NAME[status.activeBasalPattern - 1];
+        })
+        .then(() => driver.getBolusWizardSettings())
+        .then((bwzSettings) => {
+          _.assign(settings, bwzSettings);
+        })
+        .then(() => driver.getDeviceCharacteristics())
+        .then((deviceCharacteristics) => {
+          settings.pumpSerial = deviceCharacteristics.serial;
+        })
+        .then(() => driver.getDeviceString())
+        .then((deviceString) => {
+          settings.pumpModel = deviceString.model;
+        })
+        .then(() => driver.readBasalPatterns())
+        .then((schedules) => {
+          settings.basalSchedules = schedules;
+        })
+        .catch((err) => {
+          error = err;
+        })
+        // Finally
+        .then(() => driver.toggleHighSpeedMode(HighSpeedModeCommand.HIGH_SPEED_MODE.DISABLE))
+        .then(() => driver.closeConnection())
+        .then(() => driver.exitRemoteCommandMode())
+        .then(() => {
+          progress(100);
+          if (error) {
+            debug('Error getting config info: ', error.message);
+            cb(error.message, null);
+          } else {
+            data.settings = _.clone(settings);
+            cb(null, data);
+          }
+        });
+    },
+
+    fetchData(progress, data, cb) {
+      if (!isBrowser) {
+        data.fetchData = true;
+        cb(null, data);
+        return;
+      }
+      debug('in fetchData', data);
+      progress(100);
+      cb(null, data);
+    },
+
+    processData(progress, data, cb) {
+      debug('in processData');
+      cfg.builder.setDefaults({
+        deviceId: driver.pumpSession.bcnlModelAndSerial,
+      });
+      progress(100);
+      cb(null, data);
+    },
+
+    uploadData(progress, data, cb) {
+      progress(0);
+
+      const sessionInfo = {
+        deviceTags: ['insulin-pump'],
+        deviceManufacturers: ['Medtronic'],
+        deviceModel: data.settings.pumpModel,
+        deviceSerialNumber: data.settings.pumpSerial,
+        deviceId: `${data.settings.pumpModel}:${data.settings.pumpSerial}`,
+        start: sundial.utcDateString(),
+        timeProcessing: 'utc-bootstrapping',
+        tzName: cfg.timezone,
+        version: cfg.version,
+      };
+
+      const postRecords = [{
```

Comment:
I'd support continuing to use the object builder for the consistency in error messages it provides across drivers.